x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/boxplot/InternalBoxplot.java

```diff
@@ -36,39 +113,48 @@ public String value() {
             return name().toLowerCase(Locale.ROOT);
         }
 
-        double value(InternalBoxplot boxplot) {
-            switch (this) {
-                case MIN:
-                    return boxplot.getMin();
-                case MAX:
-                    return boxplot.getMax();
-                case Q1:
-                    return boxplot.getQ1();
-                case Q2:
-                    return boxplot.getQ2();
-                case Q3:
-                    return boxplot.getQ3();
-                default:
-                    throw new IllegalArgumentException("Unknown value [" + this.value() + "] in the boxplot aggregation");
-            }
+        abstract double value(InternalBoxplot boxplot);
+
+        abstract double value(TDigestState state);
+    }
+
+    /**
+     * For a given TDigest, find the "whisker" valeus, such that the upper whisker is (close to) the highest observed value less than
+     * q3 + 1.5 * IQR and the lower whisker is (close to) the lowest observed value greater than q1 - 1.5 * IQR.  Since we don't track
+     * observed values directly, this function returns the centroid according to the above logic.
+     *
+     * @param state - an initialized TDigestState representing the observed data.
+     * @return - two doubles in an array, where whiskers[0] is the lower whisker and whiskers[1] is the upper whisker.
+     */
+    public static double[] whiskers(TDigestState state) {
+        double[] results = new double[2];
+        results[0] = Double.NaN;
+        results[1] = Double.NaN;
+        if (state == null) {
+            return results;
         }
 
-        double value(TDigestState state) {
-            switch (this) {
-                case MIN:
-                    return state == null ? Double.NEGATIVE_INFINITY : state.getMin();
-                case MAX:
-                    return state == null ? Double.POSITIVE_INFINITY : state.getMax();
-                case Q1:
-                    return state == null ? Double.NaN : state.quantile(0.25);
-                case Q2:
-                    return state == null ? Double.NaN : state.quantile(0.5);
-                case Q3:
-                    return state == null ? Double.NaN : state.quantile(0.75);
-                default:
-                    throw new IllegalArgumentException("Unknown value [" + this.value() + "] in the boxplot aggregation");
+        double q3 = state.quantile(0.75);
+        double q1 = state.quantile(0.25);
+        double iqr = q3 - q1;
+        double upper = q3 + (1.5 * iqr);
+        double lower = q1 - (1.5 * iqr);
```

Comment:
I think it might be a good idea to make this a constant and give it a short explanation....